### PR TITLE
feat: use connection pooling in TPA client

### PR DESCRIPTION
**Pull request by jedinym**

### CODE REVIEW

File: src/mobster/cmd/upload/tpa.py [hunk 1]
```diff
@@ -20,8 +21,20 @@
 class TPAClient(OIDCClientCredentialsClient):
```
Comment (Allda) [inline src/mobster/cmd/upload/tpa.py:21]: TPA client is also used outside of the Mobster application and this will break existing usage. We intentionally avoided the need to use a context manager outside of the client implementation and have the logic within the client itself.

Comment (jedinym) [reply src/mobster/cmd/upload/tpa.py:21]: If I understand correctly, that should only break on release to pypi, right? Regardless, I think I can refactor to use the `__del__` magic method instead.

File: src/mobster/cmd/upload/oidc.py [hunk 2]
```diff
@@ -104,15 +118,16 @@ async def _fetch_token(self) -> None:
...
-            resp = await client.post(
-                self._auth.token_url,
-                data={
-                    "grant_type": "client_credentials",
-                    "client_id": self._auth.client_id,
-                    "client_secret": self._auth.client_secret,
-                },
-            )
+
+        assert self.client is not None
```
Comment (Allda) [inline src/mobster/cmd/upload/oidc.py:122]: Same here.

File: tests/cmd/delete/test_delete_tpa.py [hunk 3]
```diff
@@ -1,126 +0,0 @@
-from typing import Any
```
[the remark this answers is not in the document]
Comment (jedinym) [reply tests/cmd/delete/test_delete_tpa.py:1]: In my view, those tests are technical debt. They don't verify that any particular part of those commands is working, just that they are implemented by specific code.

That means that the tests fail for various reasons if the implementation changes (as in this case), even though the externally-visible behaviour has not changed. 

I'd like to focus more attention on testing at the system boundary (in this case the CLI program and how it affects TPA). I believe those tests are far better for two reasons:

One, if the behaviour changes **at the boundary**, the tests alert us that we're making a breaking change. 
Two, they are much more maintainable, for the simple reason that we never want to make breaking changes if not required, so the tests will never change, if the behaviour doesn't change.

I recommend reading [this article](https://www.tedinski.com/2018/04/10/making-tests-a-positive-influence-on-design.html), because it's way better worded than me.

---

I deleted unit tests for the delete and download commands, because all the commands are doing is visible at the boundary. That means that they can be fully covered by the integration tests. 

There are still parts of the commands, that are a bit more complicated and need unit testing. One example is the filename normalization of the download command, this is still thoroughly tested by the unit tests.
 
I deleted 200 lines of fragile code from the codebase in this PR, which is IMO a big win.
